tests/e2e/mod.rs

```diff
@@ -44,3 +44,19 @@ e2e_test!(use_version_specified_in_ruby_version_file, |dir| {
             .stderr()
     );
 });
+
+e2e_test!(use_configure_opts, |dir| {
+    dir.command()
+        .arg("install")
+        .arg("2.7.1")
+        .arg("--disable-werror")
+        .arg("--without-gmp")
+        .output();
+    dir.command().arg("local").arg("2.7.1").output();
+    let configure_opts = dir.execute_ruby(&[
+        "-r rbconfig",
+        r#"-e 'STDOUT.puts RbConfig::CONFIG["configure_args"]'"#,
+    ]);
+    assert!(configure_opts.contains("disable-werror"));
+    assert!(configure_opts.contains("without-gmp"));
+});
```

Comment:
It's hard to say how much detail we should write e2e tests. I think you only need to write tests for checking if `frum install` with some options such as `--without-gmp` and `--with-jemalloc` succeeds. So, your tests will be like the following.

```suggestion

e2e_test!(use_configure_opts, |dir| {
    dir.command()
        .arg("install")
        .arg("2.7.1")
        .arg("--disable-werror")
        .arg("--without-gmp")
        .arg("--with-jemalloc")
        .output();
});
```

Reply:
The problem is that this doesn't really test if these arguments where picked up by Ruby's `./configure` step. We could remove the code that does that and the test would still be green, that doesn't seem particularly meaningful.

Reply:
Let me think more about it.

Reply:
I found that you can get configure options doing the following.

```rust
pub fn execute_ruby(&self, args: &[&str]) -> String {
    use std::fs::File;
    use std::io::prelude::*;
    use std::io::BufReader;
    let file_path = self.dir.join("configure-options.txt");
    let mut file = std::fs::File::create(&file_path).unwrap();
    let o = Command::new("ruby")
        .current_dir(&self.dir)
        .arg("-r")
        .arg("rbconfig")
        .arg("-e")
        .arg(r#"$stdout = File.open("configure-options.txt", "w"); puts RbConfig::CONFIG["configure_args"]"#)
        .env("PATH", self.ruby_bin_path.to_str().unwrap())
        .output()
        .unwrap();

    let file = File::open(&file_path).unwrap();
    let mut buf_reader = BufReader::new(file);
    let mut contents = String::new();
    buf_reader.read_to_string(&mut contents).unwrap();
    contents
}
```

Reply:
The E2E test will be like the following.

```rust
e2e_test!(use_configure_opts, |dir| {
    dir.command()
        .arg("install")
        .arg("2.7.1")
        .arg("--disable-werror")
        .arg("--without-gmp")
        .output();
    dir.command().arg("local").arg("2.7.1").output();
    let configure_opts = dir.ruby_configure_options();
    eq_re!("--disable-werror", configure_opts);
    eq_re!("--with-openssl-dir", configure_opts);
});
```